test/test_traceradvectiondiffusion.jl

```diff
@@ -1,3 +1,74 @@
+"""
+    test_constvel1D(; kwargs...)
+
+Advect a gaussian concentration `c0(x, t)` with a constant velocity flow
+`u(x, y) = uvel` and compare the final state with
+`cfinal = c0(x - uvel * tfinal)`.
+"""
+function test_constvel1D(stepper, dt, nsteps, dev::Device=CPU())
+
+  nx, Lx = 128, 2π
+  uvel = 0.05
+  u(x) = uvel
+
+  prob = TracerAdvectionDiffusion.Problem(dev, true; nx, Lx, κ=0.0, u, dt, stepper, steadyflow=true)
+  sol, cl, vs, pr, gr = prob.sol, prob.clock, prob.vars, prob.params, prob.grid
+
+  x = gr.x
+
+  σ = 0.1
+  c0ampl = 0.1
+  c0func(x) = c0ampl * exp(-(x^2) / (2σ^2))
+
+  c0 = c0func.(x)
```

Comment:
Ah ok. That explains why when I tried `c0 = ArrayType(dev)(similar(x))` (as a preallocation) it still did not work as when it used `x` it returned an `Array` rather than a `CuArray`. Thanks very much for this! I will try the fix `x = ArrayType(dev)(gr.x)` now.